internal/backend/remote-state/cos/backend.go

```
@@ -158,27 +228,101 @@ func (b *Backend) configure(ctx context.Context) error {
 		return err
 	}
 
+	secretId := data.Get("secret_id").(string)
+	secretKey := data.Get("secret_key").(string)
+	securityToken := data.Get("security_token").(string)
+
+	// init credential by AKSK & TOKEN
+	b.credential = common.NewTokenCredential(secretId, secretKey, securityToken)
+
 	b.cosClient = cos.NewClient(
 		&cos.BaseURL{BucketURL: u},
 		&http.Client{
 			Timeout: 60 * time.Second,
 			Transport: &cos.AuthorizationTransport{
-				SecretID:  data.Get("secret_id").(string),
-				SecretKey: data.Get("secret_key").(string),
+				SecretID:     b.credential.SecretId,
+				SecretKey:    b.credential.SecretKey,
+				SessionToken: b.credential.Token,
 			},
 		},
 	)
 
-	credential := common.NewCredential(
-		data.Get("secret_id").(string),
-		data.Get("secret_key").(string),
+	// update credential if assume role exist
+	assumeRoleList := data.Get("assume_role").(*schema.Set).List()
+	if len(assumeRoleList) == 1 {
+		assumeRole := assumeRoleList[0].(map[string]interface{})
+		assumeRoleArn := assumeRole["role_arn"].(string)
+		assumeRoleSessionName := assumeRole["session_name"].(string)
+		assumeRoleSessionDuration := assumeRole["session_duration"].(int)
+		assumeRolePolicy := assumeRole["policy"].(string)
+
+		_ = b.updateCredentialWithSTS(assumeRoleArn, assumeRoleSessionName, assumeRoleSessionDuration, assumeRolePolicy)
```

the error is not checked